packages/builder/src/pages/builder/portal/manage/groups/_components/AppAddModal.svelte

```diff
@@ -0,0 +1,51 @@
+<script>
+  import { Body, ModalContent, Select } from "@budibase/bbui"
+  import { apps, groups } from "stores/portal"
+  import { roles } from "stores/backend"
+  import RoleSelect from "components/common/RoleSelect.svelte"
+
+  export let group
+
+  $: appOptions = $apps.map(app => ({
+    label: app.name,
+    value: app,
+  }))
+  $: prodAppId = selectedApp ? apps.getProdAppID(selectedApp.appId) : ""
+  $: confirmDisabled =
+    (!selectingRole && !selectedApp) || (selectingRole && !selectedRoleId)
+  let selectedApp, selectedRoleId
+  let selectingRole = false
+
+  async function appSelected() {
+    if (!selectingRole) {
+      selectingRole = true
+      await roles.fetchByAppId(prodAppId)
+      // return false to stop closing modal
+      return false
+    } else {
+      await groups.actions.addApp(group._id, prodAppId, selectedRoleId)
+    }
+  }
+</script>
+
+<ModalContent
+  onConfirm={appSelected}
+  size="M"
+  title="Add app to group"
+  confirmText="Next"
+  showSecondaryButton
+  secondaryButtonText="Back"
+  secondaryAction={() => (selectingRole = false)}
+  disabled={confirmDisabled}
+>
+  {#if !selectingRole}
+    <Body>Select the app to add to the <i>"{group.name}"</i> group.</Body>
+    <Select bind:value={selectedApp} options={appOptions} />
+  {:else}
+    <Body
+      >Pick the role this group will appear as in the <i>"{selectedApp.name}"</i
```

Comment:
`Select the role that all users in the <group name> will have for <app name>.`

Reply:
![image](https://user-images.githubusercontent.com/4407001/192340973-48187ef2-8865-477c-b572-9d74c24eba33.png)